### ISSUE TITLE
Multiple occlusion polygon support in tilemaps not functioning

### ISSUE THREAD

**Issue by ka0s420** [godotengine/godot#66877]
___
***Bugsquad note:** This issue has been confirmed several times already. No need to confirm it further.*

___

### Godot version

4.0 beta 2

### System information

win 10

### Issue description

Tilemap/tileset support for multiple occlusion polygons per tile isn't functioning. It was brought up in Godot 3x under this issue number: [#46580](https://github.com/godotengine/godot/issues/46580) which was closed to quote the thread:

"Closing as multiple collision shapes/occlusion polygons is implemented on master."

However this isn't the case.

also mentioned in this issue: #56677

use case example - I have a tile that is an empty doorway, I need both pillars of the doorway to occlude light, but not the are between them, this can only be achieved with two polygons, but in the tileset editor, creating a second polygon makes the first disappear.

### Steps to reproduce

After creating new tileset, go to rendering inspector sub-menu > Occlusion Layers > Add Element. Go over to the tileset editor, select a tile, choose to edit its occlusion layer, add one polygon making sure to close it, try to add another and the first disappears.

edit: the first polygon disappears only upon closing the second polygon shape.

**Comment by vmedea** [comment]
I was also looking at this and slightly confused. According to the documentation [TileData](https://docs.godotengine.org/en/latest/classes/class_tiledata.html) is only able to store one occlusion polygon (whereas there is support for multiple collision polygons). Unless I'm missing something, to change this would not only be an editor change, but also need a change to the underlying data structure.

**Comment by FedTheCat** [comment]
Woo! There's an issue for this! 

This has really been bugging me on one of my projects! Would love if it got some attention.

**Comment by fdragon77** [comment]
has there been any word on if this is going to be addressed? i'm just now learning godot and quickly ran into this issue when trying to make a tilemap that has 2 pieces in it. no matter what i've tried im only allowed to have one occlusion shape in the tilemap meaning that one of the two objects will just be passed through by the light in the scene.

**Comment by ka0s420** [comment]
> has there been any word on if this is going to be addressed? i'm just now learning godot and quickly ran into this issue when trying to make a tilemap that has 2 pieces in it. no matter what i've tried im only allowed to have one occlusion shape in the tilemap meaning that one of the two objects will just be passed through by the light in the scene.

Its being tracked on here, so if it gets fixed at any point or mentioned in relation to other issues it will show up here. So, as of right now, the answer is no, it's not fixed.

**Comment by groud** [comment]
I could give it a try but that's sadly a bit late for 4.3. Once https://github.com/godotengine/godot/pull/84660 is merged it should probably be a quick fix (it seems a like unrelated PR but it would cause conflicts, so I'd prefer doing it in that order).

**Comment by PerMalmberg** [comment]
Would love to see this in 4.4 now that #84660 is merged. <3

**Comment by DrJamgo** [comment]
As a work-around one can define **multiple occulsion layers** in the `Tileset` (with same layer/mask) and combine them on same tile.

**Comment by chryan** [comment]
For those building from source, I've fixed some issues in #93029.
You can get them from #95178.

**Comment by nivs1978** [comment]
> For those building from source, I've fixed some issues in #93029. You can get them from #95178.

When can we expect this to be released?

**Comment by ditiem-games** [comment]
would it be possible to also select the occluder_light_mask? In my game I noticed that I need 2 different occluder depending on the light (layer).

**Comment by Zireael07** [comment]
@ditiem-games Please make a separate issue instead of commenting on a closed one

**Comment by hasan-bro-coder** [comment]
> As a work-around one can define **multiple occulsion layers** in the `Tileset` (with same layer/mask) and combine them on same tile.

thanks a lot 
that worked for me
